src/Scim/SimpleIdServer.Scim/Helpers/RepresentationReferenceSync.cs

```diff
@@ -20,32 +17,32 @@ public class RepresentationReferenceSync : IRepresentationReferenceSync
 		private readonly ISCIMAttributeMappingQueryRepository _scimAttributeMappingQueryRepository;
 		private readonly ISCIMRepresentationCommandRepository _scimRepresentationCommandRepository;
 		private readonly ISCIMSchemaCommandRepository _scimSchemaCommandRepository;
-		private readonly IResourceTypeResolver _resourceTypeResolver;
 
 		public RepresentationReferenceSync(
 			ISCIMAttributeMappingQueryRepository scimAttributeMappingQueryRepository,
 			ISCIMRepresentationCommandRepository scimRepresentationCommandRepository,
-			ISCIMSchemaCommandRepository scimSchemaCommandRepository,
-			IResourceTypeResolver resourceTypeResolver)
+			ISCIMSchemaCommandRepository scimSchemaCommandRepository)
         {
 			_scimAttributeMappingQueryRepository = scimAttributeMappingQueryRepository;
 			_scimRepresentationCommandRepository = scimRepresentationCommandRepository;
 			_scimSchemaCommandRepository = scimSchemaCommandRepository;
-			_resourceTypeResolver = resourceTypeResolver;
 		}
 
-		public virtual IEnumerable<RepresentationSyncResult> Sync(string resourceType, SCIMRepresentation oldScimRepresentation, SCIMRepresentation newSourceScimRepresentation, string location, SCIMSchema schema, bool updateAllReferences = false, bool isScimRepresentationRemoved = false)
+        public virtual async Task<List<RepresentationSyncResult>> Sync(string resourceType, SCIMRepresentation oldScimRepresentation, SCIMRepresentation newSourceScimRepresentation, string location, SCIMSchema schema, bool updateAllReferences = true, bool isScimRepresentationRemoved = false)
```

Comment:
I rebased my code which originated from 4.0.2 using `master`, and I see that you made some changes like `false` -> `'true`.
@thabart should I change this?

Reply:
I checked the `master` branch, by default, the field `updateAllReferences` is equals to `false` and not to `true` : 

https://github.com/simpleidserver/SimpleIdServer/blob/ed077178cd645a92c4aab502b4266b3f923313fb/src/Scim/SimpleIdServer.Scim/Helpers/RepresentationReferenceSync.cs#L18

The field must be `false`